[Feat/#1] 공격 모션 구현

yuykim opened the pull request:
## 작업한 내용
- 간단한 근접 공격 모션을 구현
- 던지기 모션 구현

## 참고 사항

![55](https://github.com/user-attachments/assets/9a933341-573c-4ace-9574-1f8ac9117e44)


- X 키를 누르면 무기를 앞뒤로 움직임
> X키를 눌러서 무기가 이동하는 중에 적(enemy 태그를 가진 물체)에 닿으면 적은 2초 정지 

- C 키를 누르면 launch point로 설정해둔 곳에서 작은 원형 구체가 포물선을 그리며 날라감
> 투사체가 날아가면서 적(enemy 태그를 가진 물체)에 맞으면 적은 2초정지
> 투사체가 날아가다가 땅에 닿으면 투사체는 사라짐

- 함께 논의하고 싶은 내용:
> 투사체가 땅에 떨어져도 2초 이상 유지 되도록, 그리고 땅에 떨어진 투사체를 enemy가 밟아도 2초 정지 될 수 있게 구현하고 싶음




## 관련 이슈
- Resolved: #1

yuykim committed:
:sparkles: 근접공격 구현

x를 누르면 근접 공격

yuykim committed:
:zap: 근접 공격 업데이트

공격 성공 후, 적이 공격 범위를 나갔다가 다시 들어와야만 다시 공격이 성공하는 버그(?)를 수정

yuykim committed:
:sparkles: 던지기 모션 구현

yuykim committed:
:zap: 던지기 모션에 쿨타임 추가

3번 던지면 2초 동안 쿨타임

khlee8688 commented:
수고하셨습니다! 논의 내용에 대한 답변을 드리자면, 생존 시간 후에 투사체를 자동으로 삭제하는 부분을 지우고 적에게 맞거나 땅에 떨어지고 2초 후 destroy 하도록 해놓으면 될 거 같습니다~
PR title이나 Reviewers, Assignees, Labels는 노션에 코드 규칙 확인해보시고 수정해주세요^^

yuykim committed:
:sparkles: 던지기 모드 변환 기능 구현 :zap: 포물선으로 던지기 코드 간결하게 개선

1을 누르면 투사체를 포물선으로 던지는 모드 활설화
2를 누르면 투사체를 일직선으로 날리는 모드가 활성화됨

각각 c키를 누르면 발사

추가적으로 이전에 만들어둔 포물선으로 던지는 기능을 구현한 코드를 보다 더 간결하게 개선함

yuykim committed:
:zap: 폭발 범위 추가

폭탄이 터질때의 범위가 시각적으로 확인 가능, 해당 범위 내에 적이 있으면 데미지를 줌

yuykim committed:
:sparkles: 공격시 넉백 기능 구현

근거리 및 투척 공격시 넉백

0beanny commented:
LGTM

yuykim committed:
:sparkles: 점착폭탄 기능 구현

3번을 누르고 c를 누르면 점착폭탄이 포물선을 그리며 날라감, 스페이스바를 느르면 터짐

0beanny reviewed (approved):
LGTM

yuykim committed:
:zap: 점착 폭탄 폭발 코드 변경

기존 스페이스바를 누르면 터지는걸 b키로 변경

yuykim committed:
:zap: 레이어 수정

yuykim committed:
:bug: 레이어 이름명 수정
